app/upgrades/v15/upgrade_test.go

```diff
@@ -67,6 +78,73 @@ func (suite *UpgradeTestSuite) TestMigrateNextPoolIdAndCreatePool() {
 	suite.Require().Equal(gammPoolCreationFee, poolmanagerPoolCreationFee)
 }
 
+
+func (suite *UpgradeTestSuite) TestMigrateBalancerToStablePools() {
+	suite.SetupTest() // reset
+
+	ctx := suite.Ctx
+	gammKeeper := suite.App.GAMMKeeper
+	poolmanagerKeeper := suite.App.PoolManagerKeeper
+	// bankKeeper := suite.App.BankKeeper
+	testAccount := suite.TestAccs[0]
+
+	// Mint some assets to the accounts.
+	suite.FundAcc(testAccount, DefaultAcctFunds)
+
+	// Create the balancer pool
+	swapFee, err := sdk.NewDecFromStr("0.003")
+	exitFee, err := sdk.NewDecFromStr("0.025")
+	poolID, err := suite.App.PoolManagerKeeper.CreatePool(
+		suite.Ctx,
+		balancer.NewMsgCreateBalancerPool(suite.TestAccs[0],
+		balancer.PoolParams{
+			SwapFee: swapFee,
+			ExitFee: exitFee,
+		},
+		[]balancertypes.PoolAsset{
+			{
+				Weight: sdk.NewInt(100),
+				Token:  sdk.NewCoin("foo", sdk.NewInt(5000000)),
+			},
+			{
+				Weight: sdk.NewInt(200),
+				Token:  sdk.NewCoin("bar", sdk.NewInt(5000000)),
+			},
+		},
+		""),
+	)
+	_ = poolID
+	suite.Require().NoError(err)
+
+	balancerPool, err := gammKeeper.GetPool(suite.Ctx, poolID)
+	suite.Require().NoError(err)
+	balancerShares := balancerPool.GetTotalShares()
+
+	// join pool
+	sharesRequested := types.OneShare.MulRaw(1)
+	tokenInMaxs := sdk.Coins{}
+	_, _, err = gammKeeper.JoinPoolNoSwap(suite.Ctx, testAccount, poolID, sharesRequested, tokenInMaxs)
+	suite.Require().Equal(sharesRequested.String(), bankKeeper.GetBalance(suite.Ctx, testAccount, "gamm/pool/1").Amount.String())
```

Comment:
@p0mvn how should I think about shares minted vs. requested? They seem to be slightly off